### PR TITLE
[CCCP-403] Implement `PubKeyPreSubmitter`

### PR DESCRIPTION
**Pull request by alstjd0921**
## Description

* Implement `PubKeyPreSubmitter`

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Something else (simple changes that are not related to existing functionality or others)

# Checklist

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have made new test codes regarding to my changes.
- [ ] I have no personal secrets or credentials described on my changes.
- [ ] I have run `cargo-clippy`  and linted my code.
- [ ] My changes generate no new warnings.
- [ ] My changes passed the existing test codes.
- [ ] My changes are able to compile.

### CODE REVIEW

File: periodic/src/pub_key_presubmitter.rs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+				Ok(storage) => return storage,
+				Err(_) => {
+					tokio::time::sleep(Duration::from_millis(DEFAULT_CALL_RETRY_INTERVAL_MS)).await;
+					continue;
+				},
+			}
+		}
+	}
+
+	async fn get_n(&self) -> usize {
```
Comment (dnjscksdn98) [inline periodic/src/pub_key_presubmitter.rs:222]: 함수명이 n이라서 조금 헷갈릴거 같아서 다른 이름으로 바꾸는건 어떨까요?

Comment (dnjscksdn98) [reply periodic/src/pub_key_presubmitter.rs:222]: 이를테면 available_presubmission_len 등등

File: periodic/src/pub_key_presubmitter.rs [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+				sentry::capture_message(
+					&format!("[{}]{log_msg}", &self.bfc_client.get_chain_name()),
+					sentry::Level::Error,
+				);
+			},
+		}
+	}
+
+	/// Create n public keys.
+	async fn get_n_pub_keys(&self, n: usize) -> Vec<PublicKey> {
```
Comment (dnjscksdn98) [inline periodic/src/pub_key_presubmitter.rs:188]: 여기도?

File: periodic/src/price_source/coingecko.rs [hunk 3]
```diff
@@ -15,7 +15,6 @@ const SUB_LOG_TARGET: &str = "coingecko";
 pub struct SupportedCoin {
 	pub id: String,
 	pub symbol: String,
-	pub name: String,
```
Comment (dnjscksdn98) [inline periodic/src/price_source/coingecko.rs:18]: 이건 왜 제거된거에요?

File: periodic/src/pub_key_presubmitter.rs (hunk 1 above)
Comment (dnjscksdn98) [reply periodic/src/pub_key_presubmitter.rs:222]: 아 헷갈린 이유가 m-of-n의 n이랑 헷갈린거에요 참고로

File: periodic/src/price_source/coingecko.rs (hunk 3 above)
Comment (alstjd0921) [reply periodic/src/price_source/coingecko.rs:18]: unused field

**Review by dnjscksdn98 — APPROVED** [review]

**Pull request merged**